Applications/shapeworks/MeshCommands.cpp

```diff
@@ -1068,26 +1069,64 @@ bool WarpMesh::execute(const optparse::Values &options, SharedCommandData &share
     return false;
   }
 
+  bool multiple = static_cast<bool>(options.get("multiple"));
+
   try {
     Mesh inputMesh(inputMeshFilename);
 
     std::vector<std::string> paths;
     paths.push_back(inputPointsFilename);
-    paths.push_back(targetPointsFilename);
-    ParticleSystem particlesystem(paths);
-    Eigen::MatrixXd allPts = particlesystem.Particles();
-    Eigen::MatrixXd staticPoints = allPts.col(0);
-    Eigen::MatrixXd movingPoints = allPts.col(1);
-    int numParticles = staticPoints.rows() / 3;
-    staticPoints.resize(3, numParticles);
-    staticPoints.transposeInPlace();
-    movingPoints.resize(3, numParticles);
-    movingPoints.transposeInPlace();
-
-    MeshWarper warper;
-    warper.set_reference_mesh(inputMesh.getVTKMesh(), staticPoints);
-    Mesh output = warper.build_mesh(movingPoints);
-    output.write(targetMeshFilename);
+    if(multiple){
+      std::vector<std::string> outMeshes;
+      std::ifstream inputFile(targetPointsFilename);
+      std::string line;
+      while(inputFile >> line)
+          paths.push_back(line);
+      inputFile.close();
+
+      std::ifstream inputFileMeshes(targetMeshFilename);
+      while(inputFileMeshes >> line)
+          outMeshes.push_back(line);
+      inputFileMeshes.close();
+      
+      if (paths.size() - 1 != outMeshes.size()){
+        std::cerr << "warpmesh error: number of output point files should be equal to the number of output mesh files\n";
+        return false;
+      }
+
+      ParticleSystem particlesystem(paths);
```

Comment:
can this loop simply be used for single-file cases as well?

Reply:
Yes, I just conditioned if people want to pass directly the file paths in place of .txt files. If we are ok with that then we can just have the single command without the --multiple tag and change the tests accordingly.

Reply:
@cchriste @akenmorris what do you think about this ?